File: LivrusVirtus/KDevelop/FOXToolKit/ODBC/livrus/livrus/ConPaises.cpp
#include "resource.h"
#include "ConPaises.h"
#include "Principal.h"
#include "ConsultasSQL.h"

CfrmConPaises *frmConPaises = (CfrmConPaises *) NULL;

FXDEFMAP(CfrmConPaises) CfrmConPaisesMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmConPaises::ID_FRMCONSULTAS_BTNPESQUISAR, 
    CfrmConPaises::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmConPaises::ID_FRMCONSULTAS_BTNLIMPAR, 
    CfrmConPaises::onBtnLimpar),
  FXMAPFUNC(SEL_SELECTED,CfrmConPaises::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConPaises::onGridSelected),
  FXMAPFUNC(SEL_CHANGED,CfrmConPaises::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConPaises::onGridChanged)
};
FXIMPLEMENT(CfrmConPaises,CfrmConsultas,CfrmConPaisesMap,
  ARRAYNUMBER(CfrmConPaisesMap))

CfrmConPaises::CfrmConPaises(FXWindow* owner):
  CfrmConsultas(owner, STR_CONPAIS) {
  IniciaControles();
}

void CfrmConPaises::IniciaControles() {
  this->labDescricao->setText("País:");
  this->gridConsulta->setColumnText(1, "País");
}

void CfrmConPaises::create(){
  FXDialogBox::create();
  show(PLACEMENT_SCREEN);
}

CfrmConPaises::~CfrmConPaises(){  
}

long CfrmConPaises::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Pais('S',"", edtDescricao->getText()), STR_PAISNENC, 2);
  return true;
}

long CfrmConPaises::onBtnLimpar(FXObject*,FXSelector,void*) {
  LimparDados();
  return true;
}

long CfrmConPaises::onGridSelected(FXObject*,FXSelector,void* ptr){
  FXTableRange *escala = (FXTableRange*) ptr;
  setLabRegistros(escala->fm.row);
  return true;
}

long CfrmConPaises::onGridChanged(FXObject*,FXSelector,void* ptr){
  FXTablePos *pos = (FXTablePos*) ptr;
  setLabRegistros(pos->row);
  return true;
}

File: LivrusVirtus/FoxToolKit/FXPython/ConPaises.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConPaises(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONPAIS)  

    FXMAPFUNC(self, SEL_COMMAND, frmConPaises.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConPaises.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConPaises.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConPaises.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConPaises.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConPaises.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConPaises.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConPaises.onGridChanged)
    
    self.labDescricao.setText("Pa\xeds:")
    self.gridConsulta.setItemText(0, 2, "Pa\xeds")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Pais('S',"", 
      self.edtDescricao.getText()), Resource.STR_PAISNENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
